Add project structure & basic mqtt setup

drakeaxelrod opened the pull request:
Created src for the python portion of the project.
 Added a wrapper class for mqtt to make it easier to use.
 Created .env with python mqtt secrets, and secrets.hpp
 for ino mqtt secrets. amended gitignore to include .env
 and secrets.hpp.

drakeaxelrod committed:
Add project structure & basic mqtt setup

 Created src for the python portion of the project.
 Added a wrapper class for mqtt to make it easier to use.
 Created .env with python mqtt secrets, and secrets.hpp
 for ino mqtt secrets. amended gitignore to include .env
 and secrets.hpp

drakeaxelrod committed:
Add arrow key controls via mqtt messages

cleaned up some comments in the code that shouldn't have been there.
Added arrow key controls (functions the same as demo car). The arrow
keys send mqtt messages so the car can be controled from anywhere
if you running the program.

platisd reviewed (changes requested):
Please `gitignore` binary files :face_with_head_bandage:

drakeaxelrod committed:
Initial obstacle avoidance implementations

Obstacle avoidance using infrared sensor and ultrasonic sensor.
Emergency breaks that kick in at predefined distances. refactored
the Publisher class to a method that returns a already logged in
and connected mqtt client.
